File: asklocal/backend/internal/domain/consultant.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

// consultant object
type ConsultantProfile struct {
	// basic data for a consultant profile
	ID         uuid.UUID `json:"id" db:"id"`
	FullName   string    `json:"full_name" db:"full_name"`
	AvatarURL  string    `json:"avatar_url" db:"avatar_url"`
	Bio        string    `json:"bio" db:"bio"`
	HourlyRate float64   `json:"hourly_rate" db:"hourly_rate"`
	Rating     float64   `json:"rating" db:"rating_avg"`
	IsVerified bool      `json:"is_verified" db:"is_verified"`

	// location infor
	City    string `json:"city" db:"city_name"`
	Country string `json:"country" db:"country_code"`

	// more fields
	Languages    string `json:"languages" db:"languages"`
	ResponseTime string `json:"response_time" db:"response_time"`

	Badges []Badge `json:"badges"`

	// character's field
	Reviews   []Review  `json:"reviews"`
	Niches    []string  `json:"niches"`    // e.g. ["Foodie", "History"]
	Portfolio []string  `json:"portfolio"` // Array of Image URLs
	JoinedAt  time.Time `json:"joined_at" db:"created_at"`
}

// for mapping the review to consultant (e.g review from previous clients)
type Review struct {
	ID             uuid.UUID `json:"id" db:"id"`
	ReviewerName   string    `json:"reviewer_name" db:"reviewer_name"`
	ReviewerAvatar string    `json:"reviewer_avatar" db:"reviewer_avatar"`
	Rating         int       `json:"rating" db:"rating"`
	Comment        string    `json:"comment" db:"comment"`
	VerifiedStay   bool      `json:"verified_stay" db:"verified_stay"`
	CreatedAt      time.Time `json:"date" db:"created_at"`
}

// consultant's hobbies
type Niche struct {
	ID          int    `db:"id" json:"id"`
	Slug        string `db:"slug" json:"slug"`
	DisplayName string `db:"display_name" json:"display_name"`
}

// Badge represents a trust indicator calculated by the backend
type Badge struct {
	ID          string `json:"id"`        // e.g. "tenure_gold", "verified_id"
	IconName    string `json:"icon_name"` // Hint for frontend icon (optional)
	Title       string `json:"title"`
	Description string `json:"description"`
}

File: lokask-frontend/src/types/consultant.ts
export interface Badge {
  id: string;
  icon_name: string; // Matches Go JSON tag if using snake_case, or IconName
  title: string;
  description: string;
}

export interface Review {
  id: string;
  review_name: string;
  review_avatar: string;
  rating: number;
  comment: string;
  verified_stay: boolean;
  date: string;
}

export interface Consultant {
  id: string;
  name: string; // full name
  displayName: string; // display name
  city: string;
  country: string;
  tag: string;           // Used in the compact card
  tags: string[];        // Used in the main card
  quote: string;
  rating: number;
  helpedCount: number;
  avatarUrl: string;
  coverUrl: string;

  // Optional fields (matching the ? in your interface)
  isHighlyTrusted?: boolean;
  bio?: string;
  languages?: string[];
  responseTime?: string;
  isOnline?: boolean;
  galleryImages?: string[]; // Matches the gallery images field
  badges?: Badge[];
  reviews?: Review[];
}

export interface UpdateProfileRequest {
  full_name: string;
  display_name: string;
  city_id: number;
  quote: string;
  bio: string;
  language: string[];

  main_niche_id: number;
  tags: string[];
}